dotnet/src/SemanticKernel.Abstractions/Kernel.cs

```diff
@@ -311,36 +311,32 @@ async Task InvokeFilterOrFunctionAsync(FunctionInvocationContext context, int in
     }
 
     [Experimental("SKEXP0001")]
-    internal PromptRenderingContext? OnPromptRenderingFilter(KernelFunction function, KernelArguments arguments)
+    internal async Task<PromptRenderingContext> OnPromptRenderingAsync(
+        KernelFunction function,
+        KernelArguments arguments,
+        Func<PromptRenderingContext, Task> renderingCallback)
     {
-        PromptRenderingContext? context = null;
+        PromptRenderingContext context = new(function, arguments);
 
         if (this._promptFilters is { Count: > 0 })
         {
-            context = new(function, arguments);
-
-            for (int i = 0; i < this._promptFilters.Count; i++)
+            async Task InvokeFilterOrRenderingAsync(PromptRenderingContext context, int index = 0)
```

Comment:
No strong reason, mostly because this function will be used with recursion and I didn't want to add it on Kernel level as private since I don't expect it to be used in other place than this.